enterprise/server/cmd/goinit/main.go

```diff
@@ -0,0 +1,225 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+	"syscall"
+
+	"github.com/buildbuddy-io/buildbuddy/enterprise/server/util/vsock"
+	"github.com/buildbuddy-io/buildbuddy/enterprise/server/vmexec"
+	"github.com/buildbuddy-io/buildbuddy/server/util/log"
+	"github.com/buildbuddy-io/buildbuddy/server/util/rlimit"
+	"golang.org/x/sync/errgroup"
+	"google.golang.org/grpc"
+
+	vmxpb "github.com/buildbuddy-io/buildbuddy/proto/vmexec"
+)
+
+const (
+	logLevel = "debug"
+
+	commonMountFlags = syscall.MS_NODEV | syscall.MS_NOEXEC | syscall.MS_NOSUID
+	cgroupMountFlags = syscall.MS_NODEV | syscall.MS_NOEXEC | syscall.MS_NOSUID | syscall.MS_RELATIME
+)
+
+var (
+	cmd  = flag.String("cmd", "/sbin/getty -L ttyS0 115200 vt100", "The command to execute.")
+	path = flag.String("path", "/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin", "The path to use when executing cmd")
+	port = flag.Uint("port", vsock.DefaultPort, "The vsock port number to listen on")
+)
+
+// die logs the provided error if it is not nil and then terminates the program.
+func die(err error) {
```

Comment:
nit: what do you think about calling this die_if_err to make it clear that it's conditional?
I can also see the argument for keeping it short given how frequently it appears =P

Reply:
Yeah I was going for short / unobtrusive. Also liked the perl/php reference :P